src/OSPSuite.Core/Domain/ValueOrigin.cs

```diff
@@ -78,5 +150,24 @@ private static bool isUndefined(ValueOrigin valueOrigin)
                 valueOrigin.Method == ValueOriginDeterminationMethods.Undefined &&
                 string.IsNullOrEmpty(valueOrigin.Description);
       }
+
+      private string key
+      {
+         get
+         {
+            if (!string.IsNullOrEmpty(_key))
+               return _key;
+
+            if (isUndefined(this))
+               _key = string.Empty;
+
+            _key = new[]
+            {
+               Source.Id.ToString(), Method.Id.ToString(), Description
+            }.Where(x => !string.IsNullOrWhiteSpace(x)).ToString("-");
+
+            return _key;
```

Comment:
I did not want to have to compute the key all the time. so it's only done once and reset every time a property is changed